lib/origen/remote_manager.rb

```diff
@@ -119,8 +119,46 @@ def symlink?(path)
       end
     end
 
+    def resolve_remotes!
+      resolve_remotes
+      process_remotes
+    end
+
     private
 
+    # Process each remote
+    def process_remotes
```

Comment:
Could you comment on why this method is being added, is there nothing like this in here already?

Reply:
@ginty When I tried to use the require! method, it didn't work because it was looking only for existing remotes:

~~~ruby
def require!
      unless required?
        while updates_required?
          Origen.log.info '*******************************************************************************'
          Origen.log.info 'The following remotes need to be updated, this will now happen automatically:'
          dirty_remotes.each do |name, remote|
~~~

Because my remotes are not defined in the application.rb file, they never get created, and then cannot be dirty_remotes, which get re-assessed.  I just need them to be created.

Reply:
OK, I started trying to working out an alternative implementation which didn't introduce a new method but couldn't see an easy way either